### DISCUSSION TITLE
Support importing comments to existing Remark42 database

### DISCUSSION THREAD

**Discussion by dmitry-do** [umputun/remark42#1149] [Ideas]
Remark42 supports importing comments from Disqus, WordPress, or native backup format. E.g.,
>[Initial import from WordPress](https://github.com/umputun/remark42#initial-import-from-wordpress)

However, by the looks of it, only initial import (i.e. [clear database before importing](https://github.com/umputun/remark42/blob/e7950fcb22cca618f90db0db2218741f3847acab/backend/app/migrator/wordpress.go#L64)) is supported. This suggestion is to support import on top of existing content.

**Use case**
AS admin I WOULD LIKE TO import Disqus/WordPress comments in addition to existing Remark42 comments
SO THAT Remark42 keeps it's comments while also displaying comments imported.

**Comment by umputun** [comment]
From the provided use case I get what you want to do, but still don't see why. The import is just this - import. It happens initially as a part of the setup and to me, it is hard to imagine a use case one would setup remark42 without importing and deciding to do the import later on.

Either way, there is a way (or at least a workaround) to do what you this without any extra support - make two backup files, one for the current remark42 content, and another for imported WP/discuss/commento content. Merge them together and restore them from this file. The format of backups is straightforward json with EOL (json line) and can be easily constructed from multiple sources.

smth like this should do it (expecting to have remark42-export and wp-export prepared)

`cat wp-export | grep -v '{"version":1' >> remark42-export`

All it does is eliminating dbl meta info and merging everything else

**Reply by dmitry-do** [reply]
Thanks for the prompt reply! 
>it is hard to imagine a use case one would setup remark42 without importing and deciding to do the import later on.

Yet that's our exact use case :D 
- try Remark42 for a few weeks before migrating from another comment service,
- decide to continue with Remark42  
- realize the initial import overwrites the existing content.

Thanks for suggesting the workaround, looks like that should do the trick. 👍

**Reply by paskal** [reply]
That information will be added to the documentation in https://github.com/umputun/remark42/pull/1152 and then available [on the site](https://remark42.com/docs/backup/migration/).
